src/dotnet-grpc/Commands/CommandBase.cs

```diff
@@ -0,0 +1,343 @@
+                {
+                    Console.LogWarning(CoreStrings.LogWarningCouldNotFindFileReference, reference);
+                    continue;
+                }
+
+                resolvedReferences.Add(localItem);
+            }
+
+            return resolvedReferences;
+        }
+
+        internal string[] GlobReferences(string[] references)
+        {
+            var expandedReferences = new List<string>();
+
+            foreach (var reference in references)
+            {
+                if (IsUrl(reference))
+                {
+                    expandedReferences.Add(reference);
+                    continue;
+                }
+
+                if (Path.IsPathRooted(reference))
+                {
+                    var directoryToSearch = Path.GetPathRoot(reference);
+                    var searchPattern = reference.Substring(directoryToSearch.Length);
+
+                    expandedReferences.AddRange(Directory.GetFiles(directoryToSearch, searchPattern));
+                    continue;
+                }
+
+                try
+                {
+                    expandedReferences.AddRange(
+                        Directory.GetFiles(Project.DirectoryPath, reference)
+                            .Select(r => r.Replace(Project.DirectoryPath + Path.DirectorySeparatorChar, string.Empty)));
+                }
+                catch (DirectoryNotFoundException) { }
+            }
+
+            return expandedReferences.ToArray();
+        }
+
+        public void RemoveProtobufReference(ProjectItem protobufRef, bool removeFile)
+        {
+            Project.RemoveItem(protobufRef);
+
+            if (removeFile)
+            {
+                File.Delete(Path.IsPathRooted(protobufRef.UnevaluatedInclude) ? protobufRef.UnevaluatedInclude : Path.Combine(Project.DirectoryPath, protobufRef.UnevaluatedInclude));
+            }
+        }
+
+        public static bool IsUrl(string reference)
+        {
+            return Uri.TryCreate(reference, UriKind.Absolute, out var _) && reference.StartsWith("http", StringComparison.OrdinalIgnoreCase);
+        }
+
+        public async Task DownloadFileAsync(string url, string destination, bool dryRun = false)
+        {
+            var resolveDestination = Path.IsPathRooted(destination) ? destination : Path.Combine(Project.DirectoryPath, destination);
+
+            var contentNotModified = true;
+
+            if (!File.Exists(resolveDestination))
+            {
+                // The destination file doesn't exist so content is modified.
+                contentNotModified = false;
+
+                var destinationDirectory = Path.GetDirectoryName(resolveDestination);
+                if (!Directory.Exists(destinationDirectory))
+                {
+                    Directory.CreateDirectory(destinationDirectory);
+                }
+            }
+            else
+            {
+                using (var stream = await GetStreamAsync(url))
```

Comment:
Does this return a friendly error message if the URL is a 404, 500, etc

Reply:
HttpRequestException, we should catch that.